pbsmetrics/metrics.go

```diff
@@ -168,6 +169,7 @@ func RequestStatuses() []RequestStatus {
 		RequestStatusErr,
 		RequestStatusNetworkErr,
 		RequestStatusBlacklisted,
+		RequestStatusQueueTimeout,
```

Comment:
I'd like to start the discussion here if we want to consider a queued timeout a request. The feature is essentially a hack for nginx's queues not being able to discard messages which are meet some threshold for being expired. In other infrastructure topologies, the request might never make it to the app server. The total number of requests can be used (reasonably IMHO) for capacity planning and performance monitoring, but if we're quickly discarding these queue timeout.. should they be included? i'm personally leaning towards no, but connections and at least some server resources are used, so maybe?

If we consider this a request, this looks good. This addition adds 5 new metrics (for prometheus at least) since it's correlated to the request types (legacy, openrtb web, openrtb app, amp, and video) - which isn't a big deal when we're at around 3k metrics right now.

If we don't consider this a request, we would want to create a separate counter metric.

Reply:
I got what you mean. 
This is still a request and together with other metrics, like NetworkError and Blacklisted this additional metric is on its place. Tracking queue duration is a good idea and if we want it, then it makes sense to move queued requests related data to separate metrics structure.

Reply:
I think counting the requests is probably the way to go. In an ideal world, we want to serve all the requests in the end.

Reply:
Hans, what do you mean by counting the requests? My understanding is that we send metrics object for each request (if timeout or bad headers happened) and Grafana will count it, isn't it? 
Scott, I'm implementing a separate structure to represent only queued requests and don't mix it with regular Labels. Is this what you meant? Just to make sure.

Reply:
I was agreeing with you Veronica. :)

Reply:
Ok. Sound reasoning. I'm good with this. The only negative aspect is the queue filtering is only for video auctions right now but due to cardinality would also have metrics against the other auction types. This will add 4 unused metrics, which isn't worth a concern IMHO.